raiden-dapp/tests/unit/components/hub-list.spec.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import flushPromises from 'flush-promises';
+import Vue from 'vue';
+import Vuetify from 'vuetify';
+import { mount, Wrapper } from '@vue/test-utils';
+import { generateSuggestedPartner } from '../utils/data-generator';
+import HubList from '@/components/HubList.vue';
+import { SuggestedPartner } from 'raiden-ts/src/services/types';
+import { Address } from 'raiden-ts/src/utils/types';
+
+Vue.use(Vuetify);
+
+const addressOne = '0x123';
+const addressTwo = '0x456';
+const addressThree = '0x789';
+const suggestedPartners = [
+  generateSuggestedPartner({ address: addressOne as Address }),
+  generateSuggestedPartner({ address: addressTwo as Address }),
+  generateSuggestedPartner({ address: addressThree as Address }),
+];
+
+const createWrapper = (suggestedPartners: SuggestedPartner[] | undefined): Wrapper<HubList> => {
+  const vuetify = new Vuetify();
+
+  return mount(HubList, {
+    vuetify,
+    mocks: {
+      $t: (msg: string) => msg,
+      $raiden: {
+        getSuggestedPartners: jest.fn().mockResolvedValue(suggestedPartners),
+      },
+    },
+    propsData: {
+      tokenAddress: '0x123',
+    },
+  });
+};
+
+describe('HubList', () => {
+  test('displays spinner while fetching suggested hubs data', () => {
+    const wrapper = createWrapper([]);
+    const spinner = wrapper.find('.spinner');
+
+    expect(wrapper.vm.$data.loadingHubs).toBe(true);
+    expect(spinner.exists()).toBe(true);
+  });
+
+  test('displays message if no hubs available', async () => {
+    const wrapper = createWrapper(undefined);
+    await flushPromises();
+
+    const hubList = wrapper.find('.hub-list__no-hubs');
+
+    expect(hubList.text()).toContain('hub-list.error');
```

Comment:
Mh, should we handle errors and no hubs differently?

One example might be that no nodes are online, but the request succeeded.

Reply:
Yes, let's do this in a follow-up PR since I'm not quite sure what errors are relevant to display. So for now we just display this generic error.